hooks/badge/useMintBadge.ts

```diff
@@ -1,30 +1,29 @@
 import { ethers } from 'ethers'
 import { useState } from 'react'
-import { useContractWrite, useContractEvent, useNetwork, useAccount } from 'wagmi'
+import { useContractWrite, useContractEvent, useAccount } from 'wagmi'
 import henkakuBadge from '@/utils/abis/henkakuBadge.json'
-import { getContractAddress } from '@/utils/contractAddress'
 
 export const useMintBadge = (
+  contract: string,
+  owner: string | undefined,
   tokenID: number
 ) => {
```

Comment:
ideally, better to have contract and owner inside of hooks.
wdyt ?

Reply:
That's one thing, but the problem is that we have to call useNetwork multiple times. Not only is this error-prone, but it is also a performance concern. And that is only to get the address of the contract.

If we were to change the contract address to a method of switching by environment variables, I don't think there would be a problem with putting it in the contract hooks.

Reply:
gotcha